pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/PropertiesToMapTranslator.groovy

```diff
@@ -52,16 +57,18 @@ public class PropertiesToMapTranslator implements MethodMissingWrapper, Serializ
             argValue = args[0]
         }
 
-        if (resolveCredentials && s == "credentials") {
-            if (args.length == 1) {
-                String id = "${args[0]}"
-
-                RunWrapper currentBuild = script.getProperty("currentBuild")
+        if (resolveEnvironmentContributors) {
+            def retVal
+            if (argValue != null) {
```

Comment:
Here is the reason why we can't use `properties` as a special name in `environment` i.e. override what properties normally means. If we could find some other way of getting the `UninstantiatedDescribable` we could potentially override names of already declared steps here.